ai_summary/indico_ai_summary/controllers.py

```diff
@@ -0,0 +1,64 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2025 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+import os
+import time
+
+from flask import jsonify, request
+
+from indico.modules.events.controllers.base import RHEventBase
+from indico.modules.events.notes.util import get_scheduled_notes
+
+from indico_ai_summary.utils import cern_qwen, chunk_text, html_to_markdown, markdown_to_html
+
+
+# summarize the event
+class SummarizeEvent(RHEventBase):
+
+    CSRF_ENABLED = False
+
+    def _process_args(self):
+        print(request.view_args)
+        RHEventBase._process_args(self)
+
+    def _process(self):
+        event = self.event
+        prompt_template = request.args.get('prompt', '').strip()
+
+        notes = get_scheduled_notes(event)
+        contrib_notes_html = '\n\n'.join(note.html for note in notes if hasattr(note, 'html'))
+        meeting_notes = contrib_notes_html
+
+        if not meeting_notes.strip():
+            return jsonify({'error': "No meeting notes found from this event's contributions."}), 400
+
+        start_time = time.time()
+        cleaned_text = html_to_markdown(meeting_notes)
+        chunks = chunk_text(cleaned_text)
+        summaries = []
+        token = os.getenv('CERN_SUMMARY_API_TOKEN')
```

Comment:
I think it would make sense to set this in the plugin settings. Have a look at the [docs](https://docs.getindico.io/en/stable/plugins/getting_started/) and e.g. this [plugin](https://github.com/indico/indico-plugins/blob/master/cloud_captchas/indico_cloud_captchas/plugin.py#L32-L58) for how to implement the settings form.